packages/master-detail-layout/test/master-detail-layout.test.js

```diff
@@ -519,4 +519,104 @@ describe('vaadin-master-detail-layout', () => {
       });
     });
   });
+
+  describe('stack', () => {
+    let width, height;
+
+    before(() => {
+      width = window.innerWidth;
+      height = window.innerHeight;
+    });
+
+    afterEach(async () => {
+      await setViewport({ width, height });
+    });
+
+    describe('default', () => {
+      it('should switch from overlay to the stack mode when the stack threshold is set', async () => {
+        // Use the threshold at which the overlay mode is on by default.
+        await setViewport({ width: 350, height });
+        await nextResize(layout);
+
+        layout.stackThreshold = '400px';
+
+        expect(layout.hasAttribute('overlay')).to.be.false;
+        expect(layout.hasAttribute('stack')).to.be.true;
+        expect(getComputedStyle(detail).position).to.equal('absolute');
```

Comment:
Could add another test for viewport containment mode, where position should be fixed.

Reply:
Added a test.